packages/manager/cypress/e2e/core/account/user-permissions.spec.ts

```diff
@@ -558,4 +559,246 @@ describe('User permission management', () => {
+    cy.findByText(newUser.username).should('not.exist');
+
+    // new user should be added and shown in the user list
+    cy.findByText('Add a User')
+      .should('be.visible')
+      .should('be.enabled')
+      .click();
+
+    // confirm to add a new user
+    ui.drawer
+      .findByTitle('Add a User')
+      .should('be.visible')
+      .within(() => {
+        // an inline error message will be displayed when username or email is not specified
+        ui.buttonGroup
+          .findButtonByTitle('Add User')
+          .should('be.visible')
+          .should('be.enabled')
+          .click();
+        cy.findByText('Username is required.').should('be.visible');
+        cy.findByText('Email address is required.').should('be.visible');
+
+        // type username
+        cy.findByText('Username').click().type(`${newUser.username}{enter}`);
+
+        // an inline error message will be displayed when the email address is invalid
+        cy.findByText('Email').click().type(`not_valid_email_address{enter}`);
+        ui.buttonGroup
+          .findButtonByTitle('Add User')
+          .should('be.visible')
+          .should('be.enabled')
+          .click();
+        cy.findByText('Must be a valid Email address.').should('be.visible');
+
+        // type email address
+        cy.get('[id="email"]')
+          .click()
+          .clear()
+          .type(`${newUser.username}@test.com{enter}`);
+
+        ui.buttonGroup
+          .findButtonByTitle('Add User')
+          .should('be.visible')
+          .should('be.enabled')
+          .click();
+      });
+    // Cloud Manager passes "restricted: false" in the request payload
+    cy.wait('@addUser').then((intercept) => {
+      expect(intercept.request.body['restricted']).to.equal(newUser.restricted);
+    });
+    cy.wait('@getUsers');
+
+    // the new user is displayed in the user list
+    cy.findByText(newUser.username).should('be.visible');
+
+    // no redirect occurs
+    cy.url().should('endWith', '/users');
+  });
+
+  it('can add non full-access users with redirecting to the new user\'s "User Permission" page', () => {
+    const mockUser = accountUserFactory.build({
+      username: randomLabel(),
+      restricted: false,
+    });
+
+    const username = randomLabel();
+    const newUser = accountUserFactory.build({
+      username: username,
+      email: `${username}@test.com`,
+      restricted: true,
+    });
+
+    const mockUserGrantsUpdated = grantsFactory.build();
+    const mockUserGrants = {
+      ...mockUserGrantsUpdated,
+      global: undefined,
+    };
+
+    // TODO: Parent/Child - M3-7559 clean up when feature is live in prod and feature flag is removed.
+    mockAppendFeatureFlags({
+      parentChildAccountAccess: makeFeatureFlagData(false),
+    }).as('getFeatureFlags');
+    mockGetFeatureFlagClientstream().as('getClientStream');
+
+    mockGetUsers([mockUser]).as('getUsers');
+    mockGetUser(mockUser).as('getUser');
```

Comment:
```suggestion
    mockGetUser(mockUser);
```
It doesn't look like we're `wait`ing on this anywhere.